### ISSUE TITLE
Allow passing second type parameter to `useMap`

### ISSUE THREAD

**Issue by holubond** [uidotdev/usehooks#278]
Currently `useMap` only allows a single type parameter (for the key). It makes sense to be able to type the value as well.

Example
```ts
const myMap = useMap<string, string>();
```

**Comment by zachsents** [comment]
yeah this hook is useless from a ts perspective. why would you only want to type the keys?
